Build fails with upcoming Boost 1.89.0

In the upcoming Boost 1.89.0 release, Boost.System stub has been removed (https://github.com/boostorg/system/commit/7a495bb46d7ccd808e4be2a6589260839b0fd3a3) which causes CMake error in both devel (6e372b9a5665c8b930595af642779fa42563ab1d) and latest release (v3.0.1):

```
CMake Error at /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:141 (find_package):
  Could not find a package configuration file provided by "boost_system"
  (requested version 1.89.0) with any of the following names:

    boost_systemConfig.cmake
    boost_system-config.cmake

  Add the installation prefix of "boost_system" to CMAKE_PREFIX_PATH or set
  "boost_system_DIR" to a directory containing one of the above files.  If
  "boost_system" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  /opt/homebrew/lib/cmake/Boost-1.89.0/BoostConfig.cmake:262 (boost_find_component)
  python/CMakeLists.txt:34 (find_package)
```

---

The usage of `system` and `Boost::system` are reason for failure:

https://github.com/coal-library/coal/blob/6e372b9a5665c8b930595af642779fa42563ab1d/python/CMakeLists.txt#L34

https://github.com/coal-library/coal/blob/6e372b9a5665c8b930595af642779fa42563ab1d/python/CMakeLists.txt#L173

Fix is easier if only using a minimum of Boost 1.69[^1]. Otherwise, may need to handle with OPTIONAL_COMPONENTS and conditionals/Boost_LIBRARIES.

[^1]: https://www.boost.org/doc/libs/1_69_0/libs/system/doc/html/system.html#changes_in_boost_1_69

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,6 +16,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Remove useless COAL_WITH_CXX11_SUPPORT guard ([#688](https://github.com/coal-library/coal/pull/688))
 - Remove qhull submodule, as ubuntu 20.04 is EoL ([#704](https://github.com/coal-library/coal/pull/704))
 - Removed support for octomap < 1.8 ([#727](https://github.com/coal-library/coal/pull/727))
+- Remove direct dependency to ([#744](https://github.com/coal-library/coal/pull/744)):
+  - Boost::system
+  - Boost::chrono
+  - Boost::date_time
+  - Boost::thread
 
 ### Added
 - Added a second set of Python bindings based on nanobind ([#659](https://github.com/coal-library/coal/pull/659))

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -182,12 +182,7 @@ endif(COAL_BUILD_WITH_TRACY)
 # Required dependencies
 SET_BOOST_DEFAULT_OPTIONS()
 EXPORT_BOOST_DEFAULT_OPTIONS()
-ADD_PROJECT_DEPENDENCY(
-  Boost
-  REQUIRED
-  serialization
-  filesystem
-)
+ADD_PROJECT_DEPENDENCY(Boost REQUIRED serialization filesystem)
 if(COAL_ENABLE_LOGGING)
   ADD_PROJECT_DEPENDENCY(Boost REQUIRED log)
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -185,9 +185,6 @@ EXPORT_BOOST_DEFAULT_OPTIONS()
 ADD_PROJECT_DEPENDENCY(
   Boost
   REQUIRED
-  chrono
-  thread
-  date_time
   serialization
   filesystem
 )

--- a/python-nb/CMakeLists.txt
+++ b/python-nb/CMakeLists.txt
@@ -11,7 +11,7 @@ cmake_policy(PUSH)
 cmake_policy(SET CMP0074 NEW)
 find_package(nanobind CONFIG REQUIRED)
 cmake_policy(POP)
-find_package(Boost REQUIRED COMPONENTS system)
+find_package(Boost REQUIRED)
 
 # Nanoeigenpy
 find_package(nanoeigenpy CONFIG REQUIRED)
@@ -51,7 +51,7 @@ endif(COAL_HAS_OCTOMAP)
 nanobind_add_module(${PYTHON_LIB_NAME_V2} NB_STATIC LTO NB_SUPPRESS_WARNINGS ${PYTHON_LIB_SOURCES})
 target_link_libraries(
   ${PYTHON_LIB_NAME_V2}
-  PRIVATE ${PROJECT_NAME}::${PROJECT_NAME} Boost::system
+  PRIVATE ${PROJECT_NAME}::${PROJECT_NAME} Boost::boost
 )
 target_compile_definitions(
   ${PYTHON_LIB_NAME_V2}

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -31,7 +31,7 @@
 #  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
 #  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 #  POSSIBILITY OF SUCH DAMAGE.
-find_package(Boost REQUIRED COMPONENTS system)
+find_package(Boost REQUIRED)
 
 if(GENERATE_PYTHON_STUBS)
   include(${JRL_CMAKE_MODULES}/stubs.cmake)
@@ -170,7 +170,7 @@ endif()
 
 target_link_libraries(
   ${PYTHON_LIB_NAME}
-  PUBLIC ${PROJECT_NAME}::${PROJECT_NAME} eigenpy::eigenpy Boost::system
+  PUBLIC ${PROJECT_NAME}::${PROJECT_NAME} eigenpy::eigenpy Boost::boost
 )
 
 set_target_properties(

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -176,7 +176,7 @@ MODERNIZE_TARGET_LINK_LIBRARIES(
 
 target_link_libraries(
   ${LIBRARY_NAME}
-  PUBLIC Boost::serialization Boost::chrono Boost::filesystem
+  PUBLIC Boost::serialization Boost::filesystem
 )
 
 if(COAL_ENABLE_LOGGING)
@@ -198,10 +198,6 @@ if(COAL_BUILD_WITH_TRACY)
 endif()
 
 if(WIN32)
-  target_link_libraries(
-    ${LIBRARY_NAME}
-    INTERFACE Boost::thread Boost::date_time
-  )
   # There is an issue with MSVC 2017 and Eigen (due to std::aligned_storage).
   # See https://github.com/ceres-solver/ceres-solver/issues/481
   target_compile_definitions(

